liquidator/liquidator.js

```diff
@@ -4,25 +4,45 @@ const {
   revertWrapper
 } = require("@umaprotocol/common");
 
+const { ETH_ADDRESS } = require("./constants");
+
 class Liquidator {
   /**
    * @notice Constructs new Liquidator bot.
    * @param {Object} logger Module used to send logs.
+   * @param {Object} oneInch Module used to swap tokens on oneInch.
    * @param {Object} expiringMultiPartyClient Module used to query EMP information on-chain.
    * @param {Object} gasEstimator Module used to estimate optimal gas price with which to send txns.
    * @param {Object} votingContract DVM to query price requests.
    * @param {Object} priceFeed Module used to query the current token price.
    * @param {String} account Ethereum account from which to send txns.
+   * @param {Object} [config] Contains fields with which constructor will attempt to override defaults.
    * @param {Object} empProps Contains EMP contract state data. Expected:
    *      { crRatio: 1.5e18,
             minSponsorSize: 10e18,
             priceIdentifier: hex("ETH/BTC") }
    * @param {Object} [config] Contains fields with which constructor will attempt to override defaults.
    */
-  constructor(logger, expiringMultiPartyClient, votingContract, gasEstimator, priceFeed, account, empProps, config) {
+  constructor(
+    logger,
+    oneInch,
+    expiringMultiPartyClient,
+    votingContract,
+    gasEstimator,
+    priceFeed,
+    account,
+    empProps,
+    config
+  ) {
     this.logger = logger;
     this.account = account;
 
+    // Initial version will only allow ETH to be the reserve
+    this.reserveCurrencyAddress = ETH_ADDRESS;
```

Comment:
So this defines the currency that you will use to swap via 1inch into the token currency that you need to liquidate positions?

Reply:
Yeap

Reply:
Could we pass in the `reserveCurrencyAddress` as an optional prop in the `config` and then set the default to `ETH_ADDRESS` the same way we do other defaults? Thoughts on this pattern?

Reply:
I like this because it avoids hardcoding `this.reserveCurrencyAddress`.

Reply:
I like this approach too, but our functions are ballooning up to the point where it makes things really hard to keep track of the parameters. The recent rebase with master was particularly painful because of all the errors associated with arg/parameter additions. For example, can you spot the errors?

```javascript
// Function definition
async function run(
  logger,
  empAddress,
  oneSplitAddress,
  pollingDelay,
  errorRetries,
  errorRetriesTimeout,
  priceFeedConfig,
  liquidatorConfig,
  liquidatorOverridePrice
) { /* Logic */ }

// More code....

// In test file 1, test number 12
run(Logger, process.env.EMP_ADDRESS, oneSplitAddress, pollingDelay, errorRetriesTimeout, errorRetries, priceFeedConfig, liquidatorConfig, liquidatorOverridePrice)
```

```javascript
// In test file 1, test number 42
class Liquidator {
constructor(
    logger,
    oneInch,
    expiringMultiPartyClient,
    votingContract,
    gasEstimator,
    priceFeed,
    account,
    empProps,
    config
  ) { /* Logic */ }
}

// More code....

// In test file 1, test number 10
const liquidator = new Liquidator(spyLogger, oneInch, empProps, mockOracle, gasEstimator, priceFeedMock, accounts[0], empClient, liquidatorConfig)
```

More over, due to the nature of javascript all the arguments are an interchangable `object`, while also fails silently if you pass in the wrong sequence of parameters.

While I understand consistency is king, I **strongly** recommend we slowly change it into object destructuring notation bit by bit before introducing any more parameters

```javascript
const liquidator = new Liquidator({ 
    logger: logger,
    oneInch: oneInchMocks,
    expiringMultiPartyClient: empClient,
    votingContract: Voting,
    gasEstimator,
    priceFeed: PriceFeed,
    account: accounts[0],
    empProps: empPropsConfig,
    config: liquidatorConfig
})
```

Thoughts @nicholaspai @chrismaree ?

Reply:
> While I understand consistency is king, I strongly recommend we slowly change it into object destructuring notation bit by bit before introducing any more parameters

I agree with this 100% and this would make errors throw less silently, but I don't think we should change this in this PR as this would make this PR's diff too large and hide the "oneinch integration" main feature being introduced. What I recommend you do is follow the style in order to get this PR merged in a timely fashion and then introduce a PR that changes the construction to the pattern that you mentioned above. What's important in this PR is to be targeted with your changes and goals: you want to introduce oneinch integration into the `liquidator`, so focus on that please.

Since you have an idea in mind about how the Liquidator construction can be improved, I believe you should work on this as a separate PR. This will help other contributors to review your code faster and get all of your changes ultimately merged in sooner. Does that make sense?

Reply:
> I recommend you do is follow the style in order to get this PR merged in a timely fashion and then introduce a PR that changes the construction to the pattern that you mentioned above. What's important in this PR is to be targeted with your changes and goals: you want to introduce oneinch integration into the liquidator, so focus on that please.

Exactly! Including the change in this PR as this would make this PR's diff too big! Which is why I didn't include any of these changes in in `Poll.run` or any of the codebase apart from OneInch.

It is completely my bad including `Liquidator` as an example. I don't mean to say I want to include these changes into the codebase right now, but rather we should really consider implementing those changes if we want to include the additional configs e.g. `reserveCurrencyAddress`

Reply:
I agree with you totally @kendricktan and I think going back and implementing this as a separate change would be super useful. Overall if we have more than say 4 parameters feeding them in as named objects would be far more readable and help avoid these kinds of errors.

Let's leave this for a follow-up PR.

Reply:
@kendricktan I think the key point here is to never leave the code in an inconsistent place after PR's are merged. Be consistent until you make a follow up PR that breaks the entire construction via named objects, which we both 100% agree with.